### PR TITLE
bug fix regarding cmake set_gpu_env() for the case where using nvidia…

### PR DESCRIPTION
**Pull request by yshekel**

### CODE REVIEW

**Review by LeonHibnik — APPROVED** [review]

File: .github/changed-files.yml [hunk 1]
```diff
@@ -17,6 +17,10 @@ cpp:
   - icicle/**/*.h
   - icicle/CMakeLists.txt
   - .github/workflows/cpp_cuda.yml
+  - icicle/CMakeLists.txt
```
Comment (jeremyfelder) [inline .github/changed-files.yml:20]: Duplicate line of L#18

File: icicle/cmake/Common.cmake [hunk 2]
```diff
@@ -16,51 +16,40 @@ function(set_gpu_env)
     # each additional architecture increases the compilation time and output file size
     if(DEFINED CUDA_ARCH) # user defined arch takes priority
         set(CMAKE_CUDA_ARCHITECTURES ${CUDA_ARCH} PARENT_SCOPE)
-    elseif(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.24.0") # otherwise, use native to detect GPU arch
+    elseif(UNIX AND ${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.24.0") # otherwise, use native to detect GPU arch
```
Comment (jeremyfelder) [inline icicle/cmake/Common.cmake:19]: If we are supporting windows in general I think this is not the best place to fix the CI windows machine issue. Better to pass a "CUDA_ARCH" as a variable from within the CI job

Comment (yshekel) [reply icicle/cmake/Common.cmake:19]: I will fix

File: .github/changed-files.yml (hunk 1 above)
Comment (yshekel) [reply .github/changed-files.yml]: fixed

File: icicle/cmake/Common.cmake (hunk 2 above)
Comment (yshekel) [reply icicle/cmake/Common.cmake]: removed

**Review by jeremyfelder — APPROVED** [review]

**Pull request merged**